addons/website_sale/static/tests/interactions/snippets/dynamic_snippet_products_title.test.js

```diff
@@ -0,0 +1,64 @@
+import {
+    defineWebsiteModels,
+    setupWebsiteBuilderWithSnippet,
+} from "@website/../tests/builder/website_helpers";
+import { defineModels, fields, models, onRpc } from "@web/../tests/web_test_helpers";
+import { animationFrame, click, queryOne } from "@odoo/hoot-dom";
+import { expect, test } from "@odoo/hoot";
+
+class ProductPublicCategory extends models.Model {
+    name = fields.Char({ string: "Category name" });
+    _records = [{ id: 1, name: "All Products" }];
+}
+
+defineWebsiteModels();
+defineModels([ProductPublicCategory]);
+
+async function clickAndWait(selector) {
+    await click(selector);
+    await animationFrame();
+}
+
+async function checkNumberOfElements(actionParam, dataLabel, expectedValue, snippetEl) {
+    await clickAndWait(
+        `[data-container-title='Products'] [data-label='${dataLabel}'] button[data-action-param='${actionParam}']`
+    );
+    expect(snippetEl.dataset.numberOfElements).toBe(String(expectedValue));
+}
+
+test("for dynamic snippet products, the numberOfElements should be 2 when title is on left", async () => {
+    onRpc("/website/snippet/options_filters", async () => [
```

Comment:
To test interaction code I will add test case like other test cases does, define xml with option data-attributes applied and then will check interaction does it's job.